internal/promote/storage.go

```diff
@@ -51,6 +51,37 @@ func (pr *PackageVersion) String() string {
 	return fmt.Sprintf("%s-%s", pr.Name, pr.Version)
 }
 
+// StringWithHash method returns a string representation of the PackageVersion.
+func (pr *PackageVersion) StringWithHash(r *git.Repository, branch string) (string, error) {
```

Comment:
Let me refer to responsibility of the `PackageVersion`.

This structure is intended to keep properties of the package revision (like name or version) and would be a good candidate for the signature (SHA1). I believe the design will remain clean if you separate signature computation from the model here. I'm not convinced adding here functions accessing the local repository. 

As Shaunak suggested the signature calculation might be long, so maybe let's put it aside as a dedicated structure responsible for computing hashes for given package revisions. The output with signatures can be passed to the method responsible for opening PRs (https://github.com/elastic/elastic-package/blob/master/cmd/promote.go#L107).

Reply:
Sounds good, I'll extract it out and have it called here: https://github.com/elastic/elastic-package/blob/master/internal/promote/pull_requests.go#L75